pkg/apis/postgres-operator.crunchydata.com/v1beta1/instrumentation_types.go

```diff
@@ -91,10 +101,70 @@ type InstrumentationLogsSpec struct {
 	RetentionPeriod *Duration `json:"retentionPeriod,omitempty"`
 }
 
+type InstrumentationMetricsSpec struct {
+	// Where users can turn off built-in metrics and also provide their own
+	// custom queries.
+	// ---
+	// +optional
+	CustomQueries *InstrumentationCustomQueriesSpec `json:"customQueries,omitempty"`
+}
+
+type InstrumentationCustomQueriesSpec struct {
+	// User defined queries and metrics.
+	// ---
+	// +optional
+	Add []InstrumentationCustomQueries `json:"add,omitempty"`
+
+	// A list of built-in queries that should be removed. If all queries for a
+	// given SQL statement are removed, the SQL statement will no longer be run.
+	// ---
+	// +optional
+	Remove []string `json:"remove,omitempty"`
+}
+
+type InstrumentationCustomQueries struct {
+	// The name of this batch of queries, which will be used in naming the OTel
+	// SqlQuery receiver.
+	// ---
+	// OTel restricts component names from having whitespace, control characters,
+	// or symbols.
+	// https://github.com/open-telemetry/opentelemetry-collector/blob/main/component/identifiable.go#L23-L26
+	// +kubebuilder:validation:Pattern=`^[^\pZ\pC\pS]+$`
+	//
+	// Set a max length to keep rule costs low.
+	// +kubebuilder:validation:MaxLength=20
+	//
+	// +required
+	Name string `json:"name"`
+
+	// A ConfigMap holding the yaml file that contains the queries.
+	// ---
+	// +required
+	Queries ConfigMapKeyRef `json:"queries"`
```

Comment:
Kind of want to rename this, but not sure what it should be...

Reply:
Hmmm, they are queries, though this field is more specifically the name of a configmap that has yaml-formatted queries, right? I think `metrics.add[].queries` makes sense, but I could see some wiggle room here. What aspect of this would you want to capture in a different name?

Reply:
Yeah so you touched on it; it's that it is the name/key of the configmap, not the queries themselves that makes it feel a little off to me. That being said we've done the same thing in a number of places, so I guess it's fine.